web-animations-2/Overview.bs

```diff
@@ -2367,14 +2368,70 @@ Animation Triggers {#triggers}
 
 An <dfn export>animation trigger</dfn> is used to control the playback
 of its associated [=animation=] for time-driven animations.
-Like animations, [=animation triggers=] are associated with a [=timeline=]
-and are attached to an [=animation attachment range|attachment range=].
+An [=animation trigger=] has a {{AnimationTrigger/behavior}} attribute
+which determines the [=animation trigger|trigger=]'s effect on its
+associated [=animation=]'s playback.
 
 Issue: Should there be any effect of triggers on scroll-driven animations?
 
-### Animation Trigger Internal State ### {#trigger-state}
+There are two types of [=animation triggers=]:
+<dfn export lt='timeline-based animation trigger'>timeline-based</dfn> and
+<dfn export lt='event-based animation trigger'>event-based</dfn>. The
+type of an animation trigger is deduced from its attributes: a
+timeline-based animation trigger will have a non-empty {{AnimationTrigger/timeline}}
+attribute, while an event-based animation trigger will have non-empty
+{{AnimationTrigger/eventTarget}} and {{AnimationTrigger/eventType}} attributes.
+
+### Event-Based Animation Triggers ### {#event-based-trigger}
+
+An [=event-based animation trigger=] executes its {{AnimationTrigger/behavior}}
+when a [=qualifying event=] is dispatched.
+A <dfn for="animation trigger" export>qualifying event</dfn> is
+an {{Event!!interface}} with {{Event/type}} matching the trigger's {{AnimationTrigger/eventType}}
+and {{Event/target}} matching the trigger's {{AnimationTrigger/eventTarget}}.
+
+#### Event-Based Animation Trigger Behavior #### {#event-trigger-behavior}
+
+When a [=animation trigger/qualifying event=] is dispatched, the associated
+animation's playback is modified as follows,
+based on the trigger's {{AnimationTrigger/behavior}}
+and the current [=play state=] of the animation:
+
+<dl>
+  <dt>''animation-trigger-type/once''</dt>
+  <dd>If [=play state=] is [=play state/finished=], do nothing.
+      Otherwise, perform the actions specified for the {{Animation/play}} function.
+  </dd>
+
+  <dt>''animation-trigger-type/repeat''</dt>
+  <dd>First, if [=play state=] is [=play state/running=] or [=play state/paused=]
+      perform the actions specified for the {{Animation/cancel}} function.
```

Comment:
Again, I think it's better to reuse the procedure for `[=canceling an animation=]`, rather than reusing the Animation's method.